internal/command/test.go

```diff
@@ -1065,11 +1090,58 @@ func getEvalContextFromStates(states map[string]*TestFileState) *hcl.EvalContext
 		}
 		runCtx[state.Run.Name] = cty.ObjectVal(outputs)
 	}
-	ctx := &hcl.EvalContext{Variables: map[string]cty.Value{"run": cty.ObjectVal(runCtx)}}
+	// The output for a run block follows the format "run.<run_name>.<output_name>".
+	// The run block consists of the keyword "run" followed by the name of the run block and the output name.
+	// For example, "run.example_run.output_name".
+	if len(runCtx) > 0 {
+		ctx.Variables["run"] = cty.ObjectVal(runCtx)
+	}
+
+	varCtx := make(map[string]cty.Value)
+	for _, variable := range config.Module.Variables {
+		varCtx[variable.Name] = variable.Default
+	}
+	ctx.Variables["var"] = cty.ObjectVal(varCtx)
 
 	return ctx
 }
 
+// evaluateBlockForTest evaluates the values of "run" and "var" blocks in test files.
+// It iterates over the provider configurations defined in the given config and evaluates the hcl.Expression
+// found in the attributes of those provider configurations.
+func evaluateBlockForTest(states map[string]*TestFileState, schemas *tofu.Schemas, config *configs.Config) tfdiags.Diagnostics {
+	var diags tfdiags.Diagnostics
+	for _, provider := range config.Module.ProviderConfigs {
+		evalCtx := getEvalContextFromStates(states, config)
+		if evalCtx == nil {
+			continue
+		}
+		providerSchema := schemas.ProviderSchema(config.ProviderForConfigAddr(provider.Addr()))
+		spec := providerSchema.Provider.Block.DecoderSpec()
+
+		// If attributes are nil, then it iterates to the next ProviderConfig.
+		attributes, _ := provider.Config.JustAttributes()
+		for _, attribute := range attributes {
+			if len(attribute.Expr.Variables()) > 0 {
+				// Check if the variable root name exists in the eval context
+				if _, ok := evalCtx.Variables[attribute.Expr.Variables()[0].RootName()]; ok {
```

Comment:
to reduce nesting and improve readability it may be best to invert these if statements.